network/init.sh

```diff
@@ -65,19 +68,47 @@ $BINARY add-genesis-account $($BINARY --home $CHAIN_DIR/$CHAINID_1 keys show dem
 $BINARY add-genesis-account $($BINARY --home $CHAIN_DIR/$CHAINID_1 keys show rly1 --keyring-backend test -a) 100000000000stake  --home $CHAIN_DIR/$CHAINID_1
 $BINARY add-genesis-account $($BINARY --home $CHAIN_DIR/$CHAINID_2 keys show rly2 --keyring-backend test -a) 100000000000stake  --home $CHAIN_DIR/$CHAINID_2
 
-echo "Creating and collecting gentx..."
-$BINARY gentx val1 7000000000stake --home $CHAIN_DIR/$CHAINID_1 --chain-id $CHAINID_1 --keyring-backend test
-$BINARY gentx val2 7000000000stake --home $CHAIN_DIR/$CHAINID_2 --chain-id $CHAINID_2 --keyring-backend test
-$BINARY collect-gentxs --home $CHAIN_DIR/$CHAINID_1
-$BINARY collect-gentxs --home $CHAIN_DIR/$CHAINID_2
-
 echo "Initializing dao contract in genesis..."
 # Upload the dao contract
-$BINARY add-wasm-message store ${DAO_CONTRACT} --output json  --run-as neutron1mjk79fjjgpplak5wq838w0yd982gzkyf8fxu8u --home $CHAIN_DIR/$CHAINID_1
+$BINARY add-wasm-message store ${STAKING_CONTRACT} --output json --run-as neutron1m9l358xunhhwds0568za49mzhvuxx9ux8xafx2 --keyring-backend=test --home $CHAIN_DIR/$CHAINID_1
+$BINARY add-wasm-message store ${DAO_CONTRACT} --output json  --run-as neutron1m9l358xunhhwds0568za49mzhvuxx9ux8xafx2 --home $CHAIN_DIR/$CHAINID_1
+$BINARY add-wasm-message store ${PROPOSAL_CONTRACT} --output json  --run-as neutron1m9l358xunhhwds0568za49mzhvuxx9ux8xafx2 --home $CHAIN_DIR/$CHAINID_1
+$BINARY add-wasm-message store ${VOTING_CONTRACT} --output json  --run-as neutron1m9l358xunhhwds0568za49mzhvuxx9ux8xafx2 --home $CHAIN_DIR/$CHAINID_1
 # Instantiate the contract
-INIT_CONTRACT="$(printf '{"owner":"%s"}' "${ADMIN_ADDRESS}")"
+#INIT_CONTRACT="$(printf '{"owner":"%s"}' "${ADMIN_ADDRESS}")"
+INIT="$(printf '{"denom":"stake"}')"
+#TODO: fix quotes issue
+DAO_INIT="'""'"
 #echo "Instantiate"
-$BINARY add-wasm-message  instantiate-contract 1 "$INIT_CONTRACT" --run-as neutron1mjk79fjjgpplak5wq838w0yd982gzkyf8fxu8u --admin ${ADMIN_ADDRESS}  --label "DAO"  --home $CHAIN_DIR/$CHAINID_1
+$BINARY add-wasm-message  instantiate-contract 1 ${INIT} --run-as neutron1m9l358xunhhwds0568za49mzhvuxx9ux8xafx2 --admin ${ADMIN_ADDRESS}  --label "staking"  --home $CHAIN_DIR/$CHAINID_1
+$BINARY add-wasm-message  instantiate-contract 2 '{
```

Comment:
such a strange json formatting. is it possible to shift the whole body to the left closer to the beginning with a small indentation?

Reply:
maybe let's move the whole json to a variable as you've done with INIT?

Reply:
i already talked about this section in tg,  nvm